homeassistant/components/camera/__init__.py

```diff
@@ -991,31 +986,17 @@ async def async_handle_play_stream_service(
     ]
     other_entity_ids = list(set(entity_ids) - set(cast_entity_ids))
 
-    if cast_entity_ids:
-        await hass.services.async_call(
-            DOMAIN_MP,
-            SERVICE_PLAY_MEDIA,
-            {
-                ATTR_ENTITY_ID: cast_entity_ids,
-                **data,
-                ATTR_MEDIA_EXTRA: {
-                    "stream_type": "LIVE",
-                    "media_info": {
-                        "hlsVideoSegmentFormat": "fmp4",
-                    },
-                },
-            },
-            blocking=True,
-            context=service_call.context,
-        )
-
-    if other_entity_ids:
+    for entities, url_prefix in (
+        (cast_entity_ids, ""),
```

Comment:
The receiving media player needs to support relative URLs. Most media players don't.

Reply:
Pushed a594f9eb6e. It adds a new method `is_hass_url` to detect if it's a Home Assistant URL. This can simplify the camera service to always send the full URL